port/unix/omrsignal.c

```diff
@@ -1274,10 +1258,22 @@ registerSignalHandlerWithOS(OMRPortLibrary *portLibrary, uint32_t portLibrarySig
 		}
 	}
 
+	issueWriteBarrier();
+
 	/* Set the portLibrarySignalNo bit in signalsWithHandlers to record successful registration
 	 * of the handler. */
 	signalsWithHandlers |= portLibrarySignalNo;
```

Comment:
Seems these updates need to be atomic. Otherwise without a monitor bits could be lost.

Reply:
a monitor is always acquired before calling registerSignalHandlerWithOS.

Reply:
Is there a lock being held at this point while the flags are being modified?

Reply:
calls to `registerSignalHandlerWithOS` always acquire `registerHandlerMonitor`.